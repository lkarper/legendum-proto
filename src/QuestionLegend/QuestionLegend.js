import React from 'react';
import PropTypes from 'prop-types';

const QuestionLegend = (props) => {
    const { 
        page, 
        savedUserInput, 
    } = props;

    if (Object.keys(page).length === 0) {
        return <p>Error: Looks like something went wrong. Check your connection and try again.</p>;
    } else if (page.look_back) {
        const questionArray = page.question.split('|');
        const questionToDisplay = `${questionArray[0]}${savedUserInput[page.property_to_look_for]}${questionArray[1]}`;
        return <legend>{questionToDisplay}</legend>;
    }

    return <legend>{page.question}</legend>;
}

QuestionLegend.defaultProps = {
    page: {},
    savedUserInput: {},
};

QuestionLegend.propTypes = {
    page: PropTypes.object,
    savedUserInput: PropTypes.object,
};

export default QuestionLegend;
