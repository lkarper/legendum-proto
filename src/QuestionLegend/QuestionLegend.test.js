import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import QuestionLegend from './QuestionLegend';

describe('QuestionLegend component', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<QuestionLegend />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(<QuestionLegend />);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
