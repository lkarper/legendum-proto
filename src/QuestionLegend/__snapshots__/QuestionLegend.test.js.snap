// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionLegend component renders the UI as expected 1`] = `
<QuestionLegend
  page={Object {}}
  savedUserInput={Object {}}
>
  <p>
    Error: Looks like something went wrong. Check your connection and try again.
  </p>
</QuestionLegend>
`;
